pyirf/irf/energy_dispersion.py

```diff
@@ -110,40 +163,38 @@ def energy_dispersion_to_migration(
     new_reco_energy_edges: astropy.units.Quantity[energy]
         Reco energy edges matching the second dimension of the output
 
-    Returns:
-    --------
+    Returns
+    -------
     migration_matrix: numpy.ndarray
         Three-dimensional energy migration matrix. The third dimension
         equals the fov offset dimension of the energy dispersion matrix.
     """
-
-    migration_matrix = np.zeros((
-        len(new_true_energy_edges) - 1,
-        len(new_reco_energy_edges) - 1,
-        dispersion_matrix.shape[2],
-    ))
+    migration_matrix = np.zeros(
+        (
+            len(new_true_energy_edges) - 1,
+            len(new_reco_energy_edges) - 1,
+            dispersion_matrix.shape[2],
+        )
+    )
 
     true_energy_interpolation = resample_histogram1d(
-        dispersion_matrix,
+        dispersion_matrix / np.diff(disp_migration_edges)[np.newaxis, :, np.newaxis],
```

Comment:
You need to integrate! So multiplying with the bin_width, not divide

Reply:
right - this makes it wrong though (right is the migration matrix from the events)

![image](https://github.com/cta-observatory/pyirf/assets/17836610/1bfce769-7bec-4d15-b597-b52360a65c35)

Reply:
This instead is what happens if you multiply by the bin width instead of dividing and do not renormalize the final matrix

![image](https://github.com/cta-observatory/pyirf/assets/17836610/2c1626f6-90f5-4321-a11c-fe039f43f826)

Reply:
this is instead with the current version of this branch

![image](https://github.com/cta-observatory/pyirf/assets/17836610/46044491-ae55-4633-9ded-de2d6d676226)

Reply:
With only the one-line fix I proposed above and no other changes:
```
❯ git diff
diff --git a/pyirf/irf/energy_dispersion.py b/pyirf/irf/energy_dispersion.py
index de631e4..19e9be0 100644
--- a/pyirf/irf/energy_dispersion.py
+++ b/pyirf/irf/energy_dispersion.py
@@ -123,8 +123,10 @@ def energy_dispersion_to_migration(
         dispersion_matrix.shape[2],
     ))
 
+    migra_width = np.diff(disp_migration_edges)
+    probability = dispersion_matrix * migra_width[np.newaxis, :, np.newaxis]
     true_energy_interpolation = resample_histogram1d(
-        dispersion_matrix,
+        probability,
         disp_true_energy_edges,
         new_true_energy_edges,
         axis=0,
```

I get this on main:
![edisp_main](https://github.com/cta-observatory/pyirf/assets/5488440/507d2f66-335a-4b34-9d8f-bbbb4d5be741)


and this with the fix:
![edisp_fix](https://github.com/cta-observatory/pyirf/assets/5488440/c4ce9d97-3fdb-4053-be5b-2bb0b8c66b34)

As is to be expected by how we changed the definition of the EDISP.

Reply:
Both are actually not that good... maybe it's the interpolation. I will check how that compares to gammapy's edisp kernel method

Reply:
![gammapy_comparison](https://github.com/cta-observatory/pyirf/assets/5488440/bab9c6d7-8924-4b18-b57c-58f11b159a62)

Reply:
To be honest, I lean towards just removing this method and recommending gammapy for users interested in this.